src/OSPSuite.Presentation/Presenters/ObservedData/DataRepositoryMetaDataPresenter.cs

```diff
@@ -190,6 +200,11 @@ public void SetMolWeight(double oldMolWeightValueInDisplayUnit, double molWeight
          this.DoWithinLatch(() => { AddCommand(_editObservedDataTask.UpdateMolWeight(_allDataRepositories, molWeightValueInCoreUnit(oldMolWeightValueInDisplayUnit), molWeightValueInCoreUnit(molWeightValueInDisplayUnit))); });
       }
 
+      public Unit GetDefaultMolWeightUnit()
+      {
+         return _dimensionFactory.Dimensions.First(x => x.Name == Constants.Dimension.MOLECULAR_WEIGHT).DefaultUnit;
```

Comment:
There is a dimension by name or sthh like this

Reply:
changed to use TryGetDimension()